tools/experimental/dynamic_shapes/torchfuzz/ops_fuzzer.py

```diff
@@ -188,239 +185,65 @@ def fuzz_op(target_spec: Spec, depth, stack_size) -> tuple[str, list[Spec]]:
-        elif chosen_op == "torch.ops.aten.mul":
-            return _get_aten_mul_args_specs(target_spec)
-        elif chosen_op == "constant":
-            return _get_constant_args_specs(target_spec)
-        else:  # arg
-            return _get_arg_args_specs(target_spec)
-
-    else:
-        raise ValueError(f"Unknown target spec type: {type(target_spec)}")
-
-
-def _get_scalar_add_args_specs(target_spec: ScalarSpec) -> tuple[str, list[Spec]]:
-    """Get argument specifications for scalar_add operation using type promotion rules."""
-    # Use PyTorch's implicit type promotion rules to generate diverse input types
-    arg_specs = _get_promoted_scalar_args(target_spec.dtype)
-    return "scalar_add", arg_specs
-
-
-def _get_scalar_multiply_args_specs(target_spec: ScalarSpec) -> tuple[str, list[Spec]]:
-    """Get argument specifications for scalar_multiply operation using type promotion rules."""
-    # Use PyTorch's implicit type promotion rules to generate diverse input types
-    arg_specs = _get_promoted_scalar_args(target_spec.dtype)
-    return "scalar_multiply", arg_specs
-
-
-# Define promotion chains - types that can promote to the target
-# PyTorch promotion hierarchy (simplified):
-# - bool < int8 < int16 < int32 < int64 < float16 < float32 < float64 < complex64 < complex128
-# - uint types have limited promotion support
-_PROMOTION_CHAINS = {
-    torch.bool: [torch.bool],
-    torch.int8: [torch.bool, torch.int8],
-    torch.int16: [torch.bool, torch.int8, torch.int16],
-    torch.int32: [torch.bool, torch.int8, torch.int16, torch.int32],
-    torch.int64: [torch.bool, torch.int8, torch.int16, torch.int32, torch.int64],
-    torch.float16: [
-        torch.bool,
-        torch.int8,
-        torch.int16,
-        torch.int32,
-        torch.int64,
-        torch.float16,
-    ],
-    torch.float32: [
-        torch.bool,
-        torch.int8,
-        torch.int16,
-        torch.int32,
-        torch.int64,
-        torch.float16,
-        torch.float32,
-    ],
-    torch.float64: [
-        torch.bool,
-        torch.int8,
-        torch.int16,
-        torch.int32,
-        torch.int64,
-        torch.float16,
-        torch.float32,
-        torch.float64,
-    ],
-    torch.complex64: [
-        torch.bool,
-        torch.int8,
-        torch.int16,
-        torch.int32,
-        torch.int64,
-        torch.float16,
-        torch.float32,
-        torch.complex64,
-    ],
-    torch.complex128: [
-        torch.bool,
-        torch.int8,
-        torch.int16,
-        torch.int32,
-        torch.int64,
-        torch.float16,
-        torch.float32,
-        torch.float64,
-        torch.complex64,
-        torch.complex128,
-    ],
-}
-
-
-def _get_promoted_dtypes(target_dtype: torch.dtype) -> list[torch.dtype]:
-    """
-    Generate two dtypes that will promote to target_dtype via PyTorch's type promotion rules.
-    """
+    # Get all available operators
+    available_operators = list_operators()
```

Comment:
this function is cached right, ? we do not recreate the list each time. or do we

Reply:
or is it meant to generate a differenr order list_operators